src/structure.js

```diff
@@ -33,7 +33,7 @@ function analyzeType(value) {
       let { tree , value } = reveal(instance);
 
       if (tree.data.Type === Type && value === valueAt) {
-        return tree;
+        return graft(node.path, tree);
```

Comment:
We should definitely have a test for this codepath since it wasn't getting caught before. The symptom of not having the correct path is that `valueAt()` will be wrong.